backend/coins/eth/blockbook/blockbook.go

```diff
@@ -0,0 +1,264 @@
+
+	response, err := blockbook.httpClient.Get(reqUrl)
+	if err != nil {
+		return errp.WithStack(err)
+	}
+	defer func() { _ = response.Body.Close() }()
+	if response.StatusCode != http.StatusOK {
+		return errp.Newf("expected 200 OK, got %d", response.StatusCode)
+	}
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return errp.WithStack(err)
+	}
+
+	if err := json.Unmarshal(body, result); err != nil {
+		return errp.Newf("unexpected response from blockbook: %s", string(body))
+	}
+
+	return nil
+}
+
+func (blockbook *Blockbook) address(ctx context.Context, account common.Address, params url.Values, result interface{}) error {
+	address := account.Hex()
+
+	if err := blockbook.call(ctx, path.Join("address", address), params, result); err != nil {
+		return errp.WithStack(err)
+	}
+
+	return nil
+}
+
+// Balance implements rpc.Interface.
+func (blockbook *Blockbook) Balance(ctx context.Context, account common.Address) (*big.Int, error) {
+	result := struct {
+		Balance string `json:"balance"`
+	}{}
+
+	if err := blockbook.address(ctx, account, url.Values{}, &result); err != nil {
+		return nil, errp.WithStack(err)
+	}
+
+	balance, ok := new(big.Int).SetString(result.Balance, 10)
+	if !ok {
+		return nil, errp.Newf("could not parse balance %q", result.Balance)
+	}
+	return balance, nil
+
+}
+
+// BlockNumber implements rpc.Interface.
+func (blockbook *Blockbook) BlockNumber(ctx context.Context) (*big.Int, error) {
+	return nil, fmt.Errorf("Not yet implemented")
+}
+
+// PendingNonceAt implements rpc.Interface.
+func (blockbook *Blockbook) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
+	return 0, fmt.Errorf("Not yet implemented")
+}
+
+// prepareTransactions casts to []accounts.Transactions and removes duplicate entries and sets the
+// transaction type (send, receive, send to self) based on the account address.
+func prepareTransactions(
+	isERC20 bool,
+	blockTipHeight *big.Int,
+	isInternal bool,
+	transactions []*Tx, address common.Address) ([]*accounts.TransactionData, error) {
+
+	// TODO figure out if needed. Etherscan.go uses this to compute the num of confirmations.
+	// But numConfirmations is already returned by the API call.
+	_ = blockTipHeight
+
+	_ = isInternal // TODO figure out how to deal with internal txs.
+
+	castedTransactions := make([]*accounts.TransactionData, 0, len(transactions))
+	ours := address.Hex()
+	for _, tx := range transactions {
+		fee := coin.NewAmount(tx.FeesSat.Int)
+		timestamp := time.Unix(tx.Blocktime, 0)
+		status, err := tx.Status()
+		// TODO do not ignore unconfirmed tx
+		if status == accounts.TxStatusPending {
+			continue
+		}
+		if err != nil {
+			return nil, errp.WithStack(err)
+		}
+		from := tx.Vin[0].Addresses[0]
+		var to string
+		if len(tx.TokenTransfers) > 0 {
+			to = tx.TokenTransfers[0].To
+		} else {
+			to = tx.Vout[0].Addresses[0]
+		}
+		if ours != from && ours != to {
+			return nil, errp.New("transaction does not belong to our account")
```

Comment:
This errors on my ETH account I use for testing, because some spam tx touched it:

https://etherscan.io/tx/0x1a926b5296dc6803a8a07daeebbd6b1b6b476790af5719ec628d5eeca08027f2

This tx has a ton of ERC-1155 token transfers (NFTs?), one of which touches my address.

I am not sure so you should check, but I think the Etherscan-based account includes such transactions in the list if it originated from the account, in other words, if the ETH accounts paid the fees for it (which impacts the account balance and portfolio chart calculation). I guess it does not make sense to list "incoming" token transfers in the main ETH account, these are shown separately in the respective ERC20 accounts.